tests/unit/files/files-manager.test.ts

```diff
@@ -214,4 +338,177 @@ describe('FilesManager', () => {
+
+      // Assert - File-specific error
+      expect(fileIdForbiddenError).toBeInstanceOf(FileForbiddenError);
+      expect(fileIdForbiddenError.fileId).toBe(42);
+      expect(fileIdForbiddenError.message).toContain('Access to file with ID 42 is forbidden');
+    });
+
+    it('should properly map errors in FileErrorMapper', () => {
+      // Arrange
+      const mockRequest = new Request('http://example.com/api/upload/files/123');
+      const notFoundResponse = new Response('Not Found', { status: 404 });
+      const forbiddenResponse = new Response('Forbidden', { status: 403 });
+
+      const httpNotFoundError = new HTTPNotFoundError(notFoundResponse, mockRequest);
+      const httpForbiddenError = new HTTPForbiddenError(forbiddenResponse, mockRequest);
+
+      // Act - Create mappers with and without fileId
+      const withIdMapper = FileErrorMapper.createMapper(123);
+      const withoutIdMapper = FileErrorMapper.createMapper();
+
+      // Assert - NotFound error with fileId
+      const mappedNotFoundWithId = withIdMapper(httpNotFoundError);
+      expect(mappedNotFoundWithId).not.toBeNull();
+      expect(mappedNotFoundWithId).toBeInstanceOf(FileNotFoundError);
+      expect((mappedNotFoundWithId as FileNotFoundError).fileId).toBe(123);
+
+      // Assert - NotFound error without fileId
+      const mappedNotFoundWithoutId = withoutIdMapper(httpNotFoundError);
+      expect(mappedNotFoundWithoutId).toBeNull();
+
+      // Assert - Forbidden error with fileId
+      const mappedForbiddenWithId = withIdMapper(httpForbiddenError);
+      expect(mappedForbiddenWithId).not.toBeNull();
+      expect(mappedForbiddenWithId).toBeInstanceOf(FileForbiddenError);
+      expect((mappedForbiddenWithId as FileForbiddenError).fileId).toBe(123);
+
+      // Assert - Forbidden error without fileId
+      const mappedForbiddenWithoutId = withoutIdMapper(httpForbiddenError);
+      expect(mappedForbiddenWithoutId).not.toBeNull();
+      expect(mappedForbiddenWithoutId).toBeInstanceOf(FileForbiddenError);
+      expect((mappedForbiddenWithoutId as FileForbiddenError).fileId).toBeUndefined();
+    });
+  });
+
+  describe('error handling with interceptors', () => {
+    it('should create a custom HTTP client with error interceptors', async () => {
+      // Arrange - Spy on the create method
+      const createSpy = jest.spyOn(httpClient, 'create');
+
+      // Mock to simulate a 404 error
+      mockFetch.mockResolvedValueOnce({
+        ok: false,
+        status: 404,
+        statusText: 'Not Found',
+      });
+
+      // Act
+      try {
+        await filesManager.findOne(999);
+      } catch (_error) {
+        // We expect an error, so this is fine
+      }
+
+      // Assert
+      expect(createSpy).toHaveBeenCalled();
+
+      // Verify the interceptor was added to the client
+      const createdClient = createSpy.mock.results[0]?.value;
+      expect(createdClient?.interceptors.response).toBeDefined();
```

Comment:
This will return and verify that the interceptors' manager (to add a new one) exists, but I'm not sure what else it's supposed to check? Should we instead make sure it contains the actual interceptor?

Thing is you'll need to access the private `_handlers` property to do that :thinking: